rocksdb_replicator/replicated_db.cpp

```diff
@@ -94,10 +95,11 @@ rocksdb::Status RocksDBReplicator::ReplicatedDB::Write(
   auto end = GetCurrentTimeMs();
   logMetric(kReplicatorWriteMs, start < end ? end - start : 0, db_name_);
   
+  auto replication_mode =  common::DBConfigManager::get()->getReplicationMode(db_name_);
```

Comment:
how do we know/ensure `DBConfigManager` has loaded the config for usage here?

Reply:
static variable inside a funct will be instantiated during the fist call. And, `static` is guaranteed to be instantiated only once, thus, no race condition.

Reply:
which `static` variable are you referring to @kangnanli ?

I am asking the underlying json config loaded & parsed by `DBConfigManager`, since that's not created as part of the construction/initialization, how do we ensure it's available when we read here?

Reply:
O. I misunderstood the issue. Yes, you are right. 
`common::DBConfigManager::get()` can't guarantee the underlying json file is actually parsed and loaded into memory. That's why at the test above, it has to load it explicitly `cfgMgr->loadJsonStr(content);`. @premkumr any reason, why we want to load explicitly, instead of loading it at the DBConfigManager instantiation time at Line17-23 at dbconfig.cpp?

Reply:
During instantiation , the json config my not be available , the config fields will be populated when the config file is present and is passed on from rockstore/wc

Reply:
just in case , I've added a variable to verify data load. Without data load, the functions will respond with default values

Reply:
It's less about what we do when the config is not available, but rather can we avoid such race conditions at the first place? Typically a server should firstly load/initialize all its configs, before it start serving any runtime requests --- is that something we can achieve with rockstore/rocksplicator?

Reply:
We can handle this at the service level, where we can wait for the config file to be available just like how we now want to wait for the shardmap to be available

Reply:
We are doing a lookup for every write. any way to optimize it out?

Reply:
The map size is very small & for now it is just 1 config.  Should not be an issue. if it becomes a concern, we can have hook that will update a local variable on config change.

Reply:
chat offline with @premkumr about this --- we agree ideally we should avoid a global config and should instead pass down config via initialization, but that may require lots of refactors and plumbing config through rockstore to rocksplicator handlers, and makes dynamic config updates tricky.